common/include/kmx_file.h

```diff
@@ -341,6 +342,24 @@ struct COMP_KEYBOARD {
 
   KMX_DWORD dpBitmapOffset; // 0038 offset of the bitmaps in the file
   KMX_DWORD dwBitmapSize;   // 003C size in bytes of the bitmaps
+public:
+  /**
+   * @brief Return TRUE if keyboard is OK, otherwise FALSE
+   * Non const because the checksum gets cleared
+   *
+   * @param sz total size of keyboard structure
+   * @return KMX_BOOL
+   */
+  KMX_BOOL VerifyKeyboard(std::size_t sz);
+private:
+  /**
+   * @brief Validate checksum of the entire file
+   * Non const because the checksum gets cleared
+   *
+   * @param sz
+   * @return KMX_BOOL
+   */
+  KMX_BOOL VerifyChecksum(std::size_t sz);
```

Comment:
At present, kmx_file.h is a structural definition -- can we keep functionality separate? It could be a class inheriting from the struct in a separate .hpp -- see Core for some examples of this, such as:
 https://github.com/keymanapp/keyman/blob/be7c9c58a9e81f596a64065cbbb0a58b786d9446/core/src/keyboard.hpp#L25

and:

https://github.com/keymanapp/keyman/blob/c820fbc0e6cbd27192357a74ca0015a312cd2f8c/core/include/keyman/keyboardprocessor.h#L671-L677

Reply:
OK, didn't realize these were C also… as there are namespaces.  Will split it out.